src/jit/target.h

```diff
@@ -1384,11 +1384,15 @@ typedef unsigned short regPairNoSmall; // arm: need 12 bits
   #define REG_TAILCALL_ADDR        REG_R1
 
   // The following registers are used in emitting Enter/Leave/Tailcall profiler callbacks
-  #define REG_PROFILER_ENTER_ARG           REG_R0
-  #define RBM_PROFILER_ENTER_ARG           RBM_R0
-  #define REG_PROFILER_RET_SCRATCH         REG_R2
-  #define RBM_PROFILER_RET_SCRATCH         RBM_R2
-  #define RBM_PROFILER_RET_USED            (RBM_R0 | RBM_R1 | RBM_R2)
+  #define REG_PROFILER_ENTER_ARG1           REG_R0
+  #define RBM_PROFILER_ENTER_ARG1           RBM_R0
+  #define REG_PROFILER_ENTER_ARG2           REG_R1
+  #define RBM_PROFILER_ENTER_ARG2           RBM_R1
+  #define REG_PROFILER_RET_SCRATCH1         REG_R2
+  #define RBM_PROFILER_RET_SCRATCH1         RBM_R2
+  #define REG_PROFILER_RET_SCRATCH2         REG_R3
+  #define RBM_PROFILER_RET_SCRATCH2         RBM_R3
+  #define RBM_PROFILER_RET_USED            (RBM_R0 | RBM_R1 | RBM_R2 | RBM_R3)
```

Comment:
The fact that profiler leave call back will have return value either in r1 or r1/r2 is an ABI contract on Windows.  Changing this contract would break existing profiler on Windows.  Is there a strong reason this contract needs to be changed? If yes, then I would suggest to put under an appropriate #ifdef so that existing Windows contract is not modified.